CMakeLists.txt

```diff
@@ -223,18 +242,10 @@ if(AER_THRUST_SUPPORTED)
 			endif()
 		endif()
 		cuda_select_nvcc_arch_flags(AER_CUDA_ARCH_FLAGS ${AER_CUDA_ARCH})
-		set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -ccbin \"${CMAKE_CXX_COMPILER}\" ${AER_CUDA_ARCH_FLAGS} -DAER_THRUST_CUDA -std=c++14 -I${AER_SIMULATOR_CPP_SRC_DIR} -isystem ${AER_SIMULATOR_CPP_SRC_DIR}/third-party/headers -use_fast_math --expt-extended-lambda")
-		# We have to set SIMD flags globally because there seems to be a bug in FindCUDA which doesn't allow us to set per-file compilation flags.
-		# The implications is that users downloading the PyPi wheel package for the GPU, need to have a CPU with AVX2 support otherwise
-		# Aer will crash with: "Unknow instruction" exception.
-		# This will be fixed here: https://github.com/Qiskit/qiskit-aer/
-		if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "AMD64")
-			if(APPLE OR UNIX)
-				set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}; --compiler-options;-mfma,-mavx2")
-			elseif(MSVC)
-				set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} /arch:AVX2")
-			endif()
-		endif()
+
+		string(REPLACE ";" " "  AER_CUDA_ARCH_FLAGS_EXPAND "${AER_CUDA_ARCH_FLAGS}")
+		set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -ccbin \"${CMAKE_CXX_COMPILER}\" ${AER_CUDA_ARCH_FLAGS_EXPAND} -DAER_THRUST_CUDA -std=c++14 -I${AER_SIMULATOR_CPP_SRC_DIR} -isystem ${AER_SIMULATOR_CPP_SRC_DIR}/third-party/headers -use_fast_math --expt-extended-lambda")
```

Comment:
Have you ever experience a build failure saying something like "redefinition of std"? I get that error on one linux machine (but it runs fine on others).  It seems the flag `-std=c++14` is added twice on the `CUDA` compile commands: `nvcc .... -std=c++14 ... -std=c++14 ...`. So I removed the c++14 flag here and it compiles everything fine but fails at linkage with undefined symbols errors.

Reply:
Haven't seen that.   I'll try removing it to see if it breaks my builds.

Reply:
Builds work for me without -std-c++=14 in that spot on Ubuntu 20.04 LTS with CUDA 10.1.  There, I have to use -DCMAKE_CXX_COMPILER=g++-8.  Also it builds ok (again that way) on Windows 10 with CUDA 11. A guess is that the unresolved symbols arose by building half with one option, and another half with another option?

Reply:
I don't really know. Don't worry, I'll check what is different on this other linux machine and try to make it work.